api/db/migrations/001_initial_schema.up.sql

```diff
@@ -1,51 +1,137 @@
--- CREATE TABLE IF NOT EXISTS resumes (
---   id UUID PRIMARY KEY DEFAULT uuid_generate_v4(),
---   -- TODO cascade or restrict?
---   user_id UUID REFERENCES users ON DELETE RESTRICT,
---   url TEXT,
---   -- Reviewing state? Enum? Created, reviewing, archived?
---   created_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
---   updated_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
--- );
--- CREATE TABLE IF NOT EXISTS reviews (
---   id UUID PRIMARY KEY DEFAULT uuid_generate_v4(),
---   -- TODO cascade or restrict?
---   user_id UUID REFERENCES users ON DELETE RESTRICT,
---   resume_id UUID REFERENCES resumes ON DELETE CASCADE,
---   url TEXT,
---   -- Reviewing state? Enum?
---   created_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
---   updated_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
--- );
--- CREATE
--- OR REPLACE FUNCTION update_modified_column() RETURNS TRIGGER AS $ func $ BEGIN NEW.updated_at = NOW();
--- RETURN NEW;
--- END;
--- $ func $ LANGUAGE 'plpgsql';
--- CREATE TRIGGER update_time_users BEFORE
--- UPDATE
---   ON users FOR EACH ROW EXECUTE PROCEDURE update_modified_column();
--- CREATE TRIGGER update_time_resumes BEFORE
--- UPDATE
---   ON resumes FOR EACH ROW EXECUTE PROCEDURE update_modified_column();
--- CREATE TRIGGER update_time_reviews BEFORE
--- UPDATE
---   ON reviews FOR EACH ROW EXECUTE PROCEDURE update_modified_column();
--- -- Table for each resume reviewer, interviewer
--- -- Table for feedback?
\ No newline at end of file
+/* Enable UUID support */
+CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
+
+/* Auto timestamping function */
+CREATE OR REPLACE FUNCTION update_modified_column()
+    RETURNS TRIGGER AS
+$func$
+BEGIN
+    NEW.updated_at = now();
+    return NEW;
+END;
+$func$ LANGUAGE 'plpgsql';
+
+/* users table */
+CREATE TABLE IF NOT EXISTS users (
+    id UUID NOT NULL PRIMARY KEY DEFAULT uuid_generate_v4(),
+    email TEXT NOT NULL,
+    ccid TEXT NOT NULL,
+    program TEXT NOT NULL,
+    year SMALLINT NOT NULL,
+    given_name TEXT NOT NULL,
+    family_name TEXT NOT NULL,
+    full_name TEXT NOT NULL,
+    photo_url TEXT,
+    created_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+    updated_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+);
+
+CREATE TRIGGER update_time_users
+    BEFORE UPDATE
+    ON users
+    FOR EACH ROW EXECUTE PROCEDURE update_modified_column();
+
+/* roles table */
+CREATE TABLE roles (
+    name text NOT NULL PRIMARY KEY
+);
+
+INSERT INTO
+    roles (name)
+VALUES
+    ('reviewer'),
+    ('interviewer'),
+    ('admin');
+
+/* user_roles table */
+CREATE TABLE user_roles (
+    user_id UUID NOT NULL REFERENCES users,
+    role TEXT NOT NULL REFERENCES roles,
+    created_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+    PRIMARY KEY (user_id, role)
+);
+
+/* resume_review_states table */
+CREATE TABLE resume_review_states (
+    name TEXT NOT NULL PRIMARY KEY
+);
+
+INSERT INTO
+    resume_review_states (name)
+VALUES
+    ('creating'),
+    ('seeking_reviewer'),
+    ('reviewing'),
+    ('finished'),
+    ('cancelled');
```

Comment:
These states need refinement based on how the application will actually work.

Reply:
Yep that is roughly the idea with `creating`. Might be something we remove when we talk about it more in the meeting.

`canceled` is useful in some other scenarios too. For example imagine the scenario in which a reviewer loses their reviewer role mid review. What happens to the review? I would suggest it becomes canceled. There are other cases too.